classes/module/Module.php

```diff
@@ -205,6 +205,9 @@ abstract class ModuleCore implements ModuleInterface
     /** @var array|null used to cache module ids */
     private static $cachedModuleNames = null;
 
+    /** @var int|null Define the multistore compatibility level of the module */
+    public $multistoreCompatibility;
```

Comment:
I think it should be better to have a private attribute, and a setter (with exception if it's not an expected value) and getter.